### DISCUSSION TITLE
Data pre-processing

### DISCUSSION THREAD

**Discussion by csotiMatyas** [Project-OSRM/osrm-backend#6982] [Q&A]
Hello, is OSRM doing any pre-processing before trying to generate a route with the match api?
We are getting not the best results and I tried almost every permutation i can think of, using the raw data generated by mobile phones.

I read there are a bunch of pre-processing algorithms for: smoothing, simplifying, outlier detection, filling in missing data, temporal smoothing, etc. 

Do you do any, should be do any? If yes which ones and are there any open source / payed services to preprocess our raw data before feeding it into OSRM for better results?

**Comment by SiarheiFedartsou** [comment]
There is some very basic preprocessing, which can be enabled via tidy=true flag. See docs here https://project-osrm.org/docs/v5.24.0/api/#match-service
